api/v1alpha1/argocdrolebinding_types.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"slices"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ArgoCDRoleBindingSpec defines the desired state of ArgoCDRoleBinding
+type ArgoCDRoleBindingSpec struct {
+	Subjects      []Subject     `json:"subjects"`
+	ArgoCDRoleRef ArgoCDRoleRef `json:"argocdRoleRef"`
+}
+
+type Subject struct {
+	// +kubebuilder:validation:Enum=sso;local;role
```

Comment:
should we support only "users" and "roles" ? I don't think the distinction between sso and local is essential. local users can be defined using the Role CR, and that leave sso meaning nothing much on its own